test/fw/ptl/lib/pbs_testlib.py

```diff
@@ -13034,6 +13045,48 @@ def __init__(self, username=None, attrs={}, jobname=None):
         self.set_variable_list(self.username)
         self.set_sleep_time(100)
 
+    def set_attributes(self, a={}):
+        """
+        set attributes and custom attributes on this object.
+        custom attributes are used when converting attributes to CLI.
+        In case of Cray platform if 'Resource_List.vntype' is set
+        already the remove it and add vntype value to select statement.
+
+        :param a: Attribute dictionary
+        :type a: Dictionary
+        """
+        if isinstance(a, list):
+            a = OrderedDict(a)
+
+        self.attributes = OrderedDict(self.dflt_attributes.items() +
+                                      self.attributes.items() + a.items())
+
+        if self.platform == 'cray' or self.platform == 'craysim':
+            s = 'Resource_List.select' in a.keys()
+            v = 'Resource_List.vntype' in self.custom_attrs.keys()
```

Comment:
I'm not familiar with custom_attrs, where is it set? do we set 'Resource_List.vntype' in this dictionary by default?

Reply:
Custom attrs are nothing but the attributes other than the default attributes(specified in Job class) which test user can specify through 'attrs' parameter .  In case of Cray we are setting 'Resource_List.vntype' as a custom attribute in Jobs init function if user has not specified select statement through attrs.